### DISCUSSION TITLE
Topic on torrent ed2k Kademlia Kad Gnutella networks weaknesses...

### DISCUSSION THREAD

**Discussion by Mika3578** [ansani/Shareaza#67] [Ideas]
all good ideas welcome

**Comment by GATOQSECOMIO** [comment]
Does this client have the KAD network implemented?

**Comment by ansani** [comment]
> Does this client have the KAD network implemented?

This is the original Shareaza client (open sourced). So the answer is yes!
The implemented protocol is the full Kademlia on public networks.

**Reply by GATOQSECOMIO** [reply]
> > Does this client have the KAD network implemented?
> 
> This is the original Shareaza client (open sourced). So the answer is yes! The implemented protocol is the full Kademlia on public networks.

Thanks for the response. 
But this open issue is about the same KAD network [Support KAD #4](https://github.com/ivan386/Shareaza/issues/4). According to @ivan386 , the KAD network is not implemented in his project. Is this wrong?, sorry for the insistence.

**Reply by ansani** [reply]
> > > Does this client have the KAD network implemented?
> 
> > 
> 
> > This is the original Shareaza client (open sourced). So the answer is yes! The implemented protocol is the full Kademlia on public networks.
> 
> 
> 
> Thanks for the response. 
> 
> But this open issue is about the same KAD network [Support KAD #4](https://github.com/ivan386/Shareaza/issues/4). According to @ivan386 , the KAD network is not implemented in his project. Is this wrong? sorry for the insistence.

Oh ok, now I understood what do you mean. As per official Shareaza ticket (https://shareaza.sourceforge.net/phpbb/viewtopic.php?f=2&t=813) the client supports only the KAD cache and host file (the .dat list).

Please open a specific ticket so I plan it into next releases.
